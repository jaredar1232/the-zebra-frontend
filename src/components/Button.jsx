import React from "react";
import { ReactComponent as IconArrowDown } from "../assets/IconArrowDown.svg";
import { ReactComponent as IconBoltWhite } from "../assets/IconBoltWhite.svg";

/**
 * Type determines color
 * Text determines icon
 */
const Button = (props) => {
  if (props.type === 0 || props.type === 1) {
    return (
      <a href={props.data.link} target="_blank" rel="noopener noreferrer">
        {props.data.link_copy === "Buy Online Now" ? (
          <IconBoltWhite />
        ) : (
          <IconArrowDown />
        )}
        {props.data.link_text ? props.data.link_text : props.data.link_copy}
      </a>
    );
  } else if (props.type === 3) {
    return (
      <a href={props.data.link} target="_blank" rel="noopener noreferrer">
        {props.data.link_copy === "Buy Online Now" ? (
          <IconBoltWhite />
        ) : (
          <IconArrowDown />
        )}
        {props.data.link_text ? props.data.link_text : props.data.link_copy}
      </a>
    );
  } else if (props.type === 2) {
    return null;
  }
};

export default Button;
