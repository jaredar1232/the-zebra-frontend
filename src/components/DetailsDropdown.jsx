import React from "react";

const DetailsDropdown = (props) => {
  if (props.details) {
    return (
      <div>
        <h2>Why {props.name}?</h2>
        <div />
        <p>{props.details}</p>
      </div>
    );
  }
  return null;
};

export default DetailsDropdown;
