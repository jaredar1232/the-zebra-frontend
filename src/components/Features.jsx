import React from "react";
import { ReactComponent as IconFeaturesBuyOnline } from "../assets/IconFeaturesBuyOnline.svg";
import { ReactComponent as IconFeaturesLocalAgent } from "../assets/IconFeaturesLocalAgent.svg";
import { ReactComponent as IconFeaturesMobile } from "../assets/IconFeaturesMobile.svg";
import { ReactComponent as IconFeaturesPaperless } from "../assets/IconFeaturesPaperless.svg";
import { ReactComponent as IconFeaturesNewCarReplacement } from "../assets/IconFeaturesNewCarReplacement.svg";
import { ReactComponent as IconFeaturesImmediateCoverage } from "../assets/IconFeaturesImmediateCoverage.svg";
import { ReactComponent as IconFeaturesAutomaticPayment } from "../assets/IconFeaturesAutomaticPayment.svg";
import { ReactComponent as IconFeaturesOnlineIdCards } from "../assets/IconFeaturesOnlineIdCards.svg";
import { ReactComponent as IconFeaturesSR22 } from "../assets/IconFeaturesSR22.svg";
import { ReactComponent as IconFeaturesRoadside } from "../assets/IconFeaturesRoadside.svg";
import { ReactComponent as IconFeaturesRideshare } from "../assets/IconFeaturesRideshare.svg";
import { ReactComponent as IconFeaturesRentersBundle } from "../assets/IconFeaturesRentersBundle.svg";
import { ReactComponent as IconFeaturesRateLock } from "../assets/IconFeaturesRateLock.svg";
import { ReactComponent as IconFeaturesOnlineServicing } from "../assets/IconFeaturesOnlineServicing.svg";
import { ReactComponent as IconFeaturesNonOwner } from "../assets/IconFeaturesNonOwner.svg";
import { ReactComponent as IconFeaturesNoCreditCheck } from "../assets/IconFeaturesNonOwner.svg";
import { ReactComponent as IconFeaturesLowDownPayment } from "../assets/IconFeaturesLowDownPayment.svg";
import { ReactComponent as IconFeaturesDriverExclusion } from "../assets/IconFeaturesDriverExclusion.svg";
import { ReactComponent as IconFeaturesCreditCard } from "../assets/IconFeaturesCreditCard.svg";
import { ReactComponent as IconFeatures247Support } from "../assets/IconFeatures247Support.svg";

const Features = (props) => {
  // Takes in a features array. Returns an array of feature elements
  const buildFeatures = (featureNamesArray) => {
    let featuresArray = [];

    // TODO: REFACTOR
    /** 
    I know... This "seems" hideous. Its just a for loop
    looping over an array of feature names. Inside is a 
    20 statement ternary. If a name matches a ternary then 
    the svg component is pushed into the features array. 
    Will refactor if time allows. Maybe a string to JSX library?
    */
    for (let i = 0; i < featureNamesArray.length; i++) {
      //   console.log(featureNamesArray[i].icon);
      let iconName = featureNamesArray[i].icon;

      //   featuresArray.push(JSON.parse(`<${iconName} />`));

      //   let pushable =
      //     iconName === "IconFeaturesBuyOnline" ? (
      //       <IconFeaturesBuyOnline />
      //     ) : featureNamesArray[i].icon === "IconFeaturesLocalAgent" ? (
      //       <IconFeaturesLocalAgent />
      //     ) : featureNamesArray[i].icon === "IconFeaturesMobile" ? (
      //       <IconFeaturesMobile />
      //     ) : featureNamesArray[i].icon === "IconFeaturesPaperless" ? (
      //       <IconFeaturesPaperless />
      //     ) : featureNamesArray[i].icon === "IconFeaturesNewCarReplacement" ? (
      //       <IconFeaturesNewCarReplacement />
      //     ) : featureNamesArray[i].icon === "IconFeaturesImmediateCoverage" ? (
      //       <IconFeaturesImmediateCoverage />
      //     ) : featureNamesArray[i].icon === "IconFeaturesAutomaticPayment" ? (
      //       <IconFeaturesAutomaticPayment />
      //     ) : featureNamesArray[i].icon === "IconFeaturesOnlineIdCards" ? (
      //       <IconFeaturesOnlineIdCards />
      //     ) : featureNamesArray[i].icon === "IconFeaturesSR22" ? (
      //       <IconFeaturesSR22 />
      //     ) : featureNamesArray[i].icon === "IconFeaturesRoadside" ? (
      //       <IconFeaturesRoadside />
      //     ) : featureNamesArray[i].icon === "IconFeaturesRideshare" ? (
      //       <IconFeaturesRideshare />
      //     ) : featureNamesArray[i].icon === "IconFeaturesRentersBundle" ? (
      //       <IconFeaturesRentersBundle />
      //     ) : featureNamesArray[i].icon === "IconFeaturesRateLock" ? (
      //       <IconFeaturesRateLock />
      //     ) : featureNamesArray[i].icon === "IconFeaturesOnlineServicing" ? (
      //       <IconFeaturesOnlineServicing />
      //     ) : featureNamesArray[i].icon === "IconFeaturesNonOwner" ? (
      //       <IconFeaturesNonOwner />
      //     ) : featureNamesArray[i].icon === "IconFeaturesNoCreditCheck" ? (
      //       <IconFeaturesNoCreditCheck />
      //     ) : featureNamesArray[i].icon === "IconFeaturesLowDownPayment" ? (
      //       <IconFeaturesLowDownPayment />
      //     ) : featureNamesArray[i].icon === "IconFeaturesDriverExclusion" ? (
      //       <IconFeaturesDriverExclusion />
      //     ) : featureNamesArray[i].icon === "IconFeaturesCreditCard" ? (
      //       <IconFeaturesCreditCard />
      //     ) : featureNamesArray[i].icon === "IconFeatures247Support" ? (
      //       <IconFeatures247Support />
      //     ) : null;

      //   featuresArray.push(pushable);

      if (iconName === "IconFeaturesBuyOnline") {
        featuresArray.push(<IconFeaturesBuyOnline />);
      }
      if (featureNamesArray[i].icon === "IconFeaturesLocalAgent") {
        featuresArray.push(<IconFeaturesLocalAgent />);
      }
      if (featureNamesArray[i].icon === "IconFeaturesMobile") {
        featuresArray.push(<IconFeaturesMobile />);
      }
      if (featureNamesArray[i].icon === "IconFeaturesPaperless") {
        featuresArray.push(<IconFeaturesPaperless />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesNewCarReplacement") {
        featuresArray.push(<IconFeaturesNewCarReplacement />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesImmediateCoverage") {
        featuresArray.push(<IconFeaturesImmediateCoverage />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesAutomaticPayment") {
        featuresArray.push(<IconFeaturesAutomaticPayment />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesOnlineIdCards") {
        featuresArray.push(<IconFeaturesOnlineIdCards />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesSR22") {
        featuresArray.push(<IconFeaturesSR22 />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesRoadside") {
        featuresArray.push(<IconFeaturesRoadside />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesRideshare") {
        featuresArray.push(<IconFeaturesRideshare />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesRentersBundle") {
        featuresArray.push(<IconFeaturesRentersBundle />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesRateLock") {
        featuresArray.push(<IconFeaturesRateLock />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesOnlineServicing") {
        featuresArray.push(<IconFeaturesOnlineServicing />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesNonOwner") {
        featuresArray.push(<IconFeaturesNonOwner />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesNoCreditCheck") {
        featuresArray.push(<IconFeaturesNoCreditCheck />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesLowDownPayment") {
        featuresArray.push(<IconFeaturesLowDownPayment />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesDriverExclusion") {
        featuresArray.push(<IconFeaturesDriverExclusion />);
      }

      if (featureNamesArray[i].icon === "IconFeaturesCreditCard") {
        featuresArray.push(<IconFeaturesCreditCard />);
      }

      if (featureNamesArray[i].icon === "IconFeatures247Support") {
        featuresArray.push(<IconFeatures247Support />);
      }
    }
    return featuresArray;
  };

  return <div> {buildFeatures(props.features)}</div>;
};

export default Features;
