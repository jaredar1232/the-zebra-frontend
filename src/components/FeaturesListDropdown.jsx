import React from "react";

const FeaturesListDropdown = (props) => {
  if (props.features) {
    return (
      <div>
        <h2>Features</h2>
        <div />
        <p dangerouslySetInnerHTML={{ __html: props.features }}></p>
      </div>
    );
  } else {
    return null;
  }
};

export default FeaturesListDropdown;
