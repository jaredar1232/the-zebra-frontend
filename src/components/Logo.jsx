import React from "react";
import image from "../assets/logo-carrier-sm-default.png";

const Logo = () => {
  return (
    <div className="logo-container" data-testid="logo-test-id">
      <img className="logo" alt="" src={image}></img>
    </div>
  );
};

export default Logo;
