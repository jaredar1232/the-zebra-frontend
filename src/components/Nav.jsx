import toMinutes from "../helpers/toMinutes";

// Helpers:
// toMinutes takes in milliseconds as a number. Returns time and as a string

const Nav = (props) => {
  return (
    <div className="nav-container">
      <div className="nav__query-info">
        <div className="nav__query-info-content">
          You just compared
          <b className="nav__query-info-content--bold">{` ${props.carriersSearched} rates`}</b>
          {` in ${toMinutes(props.searchTime)}!`}
        </div>
      </div>

      <form className="nav__form" id="sortform">
        <label>
          Sort by:
          <select
            className="nav__form-selector"
            form="sortform"
            value={props.formValue}
            onChange={(e) => props.handleFormChange(e)}
          >
            <option value="highest">Price: Highest First</option>
            <option value="lowest">Price: Lowest First</option>
            <option value="best">Rating: Highest First</option>
            <option value="worst">Rating: Lowest First</option>
            <option value="alphabeticalA">{`Alphabetical (A-Z)`}</option>
            <option value="alphabeticalZ">{`Alphabetical (Z-A)`}</option>
          </select>
        </label>
      </form>
    </div>
  );
};

export default Nav;
