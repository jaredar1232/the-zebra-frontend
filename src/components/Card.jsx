import React from "react";
import Stars from "./CardComponents/Stars";
import Logo from "./CardComponents/Logo";
import Features from "./CardComponents/Features";
import Button from "./CardComponents/Button";
import VerifiedMark from "./CardComponents/VerifiedMark";
import FeaturesListDropdown from "./CardComponents/FeaturesListDropdown";
import DetailsDropdown from "./CardComponents/DetailsDropdown";
import Price from "./CardComponents/Price";
import { ReactComponent as Chevron } from "../assets/IconChevronDown.svg";

export default class Card extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isExpanded: false,
    };
  }

  // Toggles isExpanded state
  handleAccordianClick = () => {
    /**
     * I originally wanted to toggle the accordian with clicks to certain
     * parts of the parent but that will take more set up to filter each
     * element clicked
     */

    this.setState((prevState) => ({
      isExpanded: !prevState.isExpanded,
    }));
  };

  render() {
    let carrier = this.props.aCardData;

    return (
      <div className="card">
        <VerifiedMark isVerified={carrier.tag} />

        <div className="card-topfold-container">
          <div className="card-topfold-subcontainer-1">
            <Logo />
          </div>

          <div className="card-topfold-subcontainer-2">
            <h1 className="carrier-name" data-testid="carriername-test-id">
              {carrier.name}
            </h1>
            <Stars stars={carrier.stars} />
            <Features features={carrier.features} />
            {carrier.tagline ? (
              <div className="carrier-tagline" data-testid="tagline-test-id">
                {carrier.tagline}
              </div>
            ) : null}
          </div>

          <div className="card-topfold-subcontainer-3">
            <div className="card-tag" data-testid="tag-test-id">
              {carrier.tag}
            </div>
            <Price rate={carrier.rate} type={carrier.type} />
            <Button data={carrier.action} type={carrier.type} />
          </div>
        </div>

        <div
          className={
            this.state.isExpanded ? "dropdown-shown" : "dropdown-hidden"
          }
        >
          <FeaturesListDropdown features={carrier.features_html} />
          <DetailsDropdown name={carrier.name} details={carrier.detail_body} />
        </div>

        <button
          className="chevron-button"
          onClick={(e) => this.handleAccordianClick(e)}
        >
          <Chevron
            data-testid="chevron-test-id"
            className={this.state.isExpanded ? "chevron-up" : "chevron-down"}
          />
        </button>
      </div>
    );
  }
}
