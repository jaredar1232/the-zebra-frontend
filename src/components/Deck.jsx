import Card from "./Card";
import sortData from "../helpers/sortData";

// Helpers:
// sortData takes in string keyword and unstorted data array. Returns sorted data array

// Container for mapping and holding all carrier cards
const Deck = (props) => {
  return (
    <div className="deck__card-container">
      {sortData(props.formValue, props.carrierCardData).map((aCardData) => (
        <Card aCardData={aCardData} key={aCardData.name} />
      ))}
    </div>
  );
};

export { Deck, sortData };
