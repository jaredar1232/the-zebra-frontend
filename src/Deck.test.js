import React from "react";
import renderer from "react-test-renderer";
import carrierCardData from "../TestData";
import { Deck, sortData } from "./components/Deck";

describe("Deck Unit Testing", () => {
  describe("test decks sortdata helper", () => {
    test("deck has a sortdata helper", () => {
      expect(sortData).toBeDefined();
    });

    test("initial card should be low card", () => {
      expect(carrierCardData[0].name).toMatch("Low Card");
    });

    test("initial card should not be high card", () => {
      expect(carrierCardData[0].name).not.toMatch("High Card");
    });

    // Test price keywords
    test("sort highest price returns high card first ", () => {
      expect(sortData("highest", carrierCardData)[0].name).toMatch("High Card");
    });

    test("sort highest price should not return low card first ", () => {
      expect(sortData("highest", carrierCardData)[0].name).not.toMatch(
        "Low Card"
      );
    });

    test("sort lowest price returns low card first ", () => {
      expect(sortData("lowest", carrierCardData)[0].name).toMatch("Low Card");
    });

    test("sort lowest price should not return high card first ", () => {
      expect(sortData("lowest", carrierCardData)[0].name).not.toMatch(
        "High Card"
      );
    });

    // Test rating keywords
    test("sort best rating returns high card first ", () => {
      expect(sortData("best", carrierCardData)[0].name).toMatch("High Card");
    });

    test("sort best rating should not return low card first ", () => {
      expect(sortData("best", carrierCardData)[0].name).not.toMatch("Low Card");
    });

    test("sort worst rating returns low card first ", () => {
      expect(sortData("worst", carrierCardData)[0].name).toMatch("Low Card");
    });

    test("sort worst rating should not return high card first ", () => {
      expect(sortData("worst", carrierCardData)[0].name).not.toMatch(
        "High Card"
      );
    });

    // Test alphabetical keywords: lowcard always first
    test("sort alphabetical a-z returns low card first ", () => {
      expect(sortData("alphabeticalA", carrierCardData)[0].name).toMatch(
        "Low Card"
      );
    });

    test("sort alphabetical a-z does not return high card first ", () => {
      expect(sortData("alphabeticalA", carrierCardData)[0].name).not.toMatch(
        "High Card"
      );
    });

    test("sort alphabetical z-a returns low card first ", () => {
      expect(sortData("alphabeticalZ", carrierCardData)[0].name).toMatch(
        "Low Card"
      );
    });

    test("sort alphabetical z-a does not return high card first ", () => {
      expect(sortData("alphabeticalZ", carrierCardData)[0].name).not.toMatch(
        "High Card"
      );
    });
  });

  describe("test decks render", () => {
    test("deck renders correctly", () => {
      const formValue = "best";
      const tree = renderer
        .create(
          <Deck carrierCardData={carrierCardData} formValue={formValue} />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
