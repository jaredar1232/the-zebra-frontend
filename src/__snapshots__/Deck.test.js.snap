// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck Unit Testing test decks render deck renders correctly 1`] = `
<div
  className="deck__card-container"
>
  <div
    className="card"
    data-testid="card-test-id"
  >
    <div
      className="card__verified-mark"
    >
      <svg>
        IconStarWhite.svg
      </svg>
    </div>
    <div
      className="card__topfold-container"
    >
      <div
        className="card__topfold-subcontainer--1"
      >
        <div
          className="card__logo-container"
        >
          <img
            alt=""
            className="card__logo"
            src="logo-carrier-sm-default.png"
          />
        </div>
      </div>
      <div
        className="card__topfold-subcontainer--2"
      >
        <h1
          className="card__carrier-name"
        >
          High Card
        </h1>
        <div
          className="card__star-icons"
        >
          <svg>
            IconStarGold.svg
          </svg>
          <svg>
            IconStarGold.svg
          </svg>
          <svg>
            IconStarGold.svg
          </svg>
          <svg>
            IconStarGold.svg
          </svg>
          <svg>
            IconStarGold.svg
          </svg>
        </div>
        <div
          className="card__feature-icons"
        >
          <svg>
            IconFeaturesLocalAgent.svg
          </svg>
          <svg>
            IconFeaturesMobile.svg
          </svg>
        </div>
        <div
          className="card__carrier-tagline"
        >
          Austin, TX drivers could save up to 25% with Elephant®
        </div>
      </div>
      <div
        className="card__topfold-subcontainer--3"
      >
        <div
          className="card__verified-tag"
        >
          Verified Price
        </div>
        <div
          className="card__price"
        >
          <span
            className="card__price-dollar-sign"
          >
            $
          </span>
          NaN
           
          <span
            className="card__price-month"
          >
            / mo
          </span>
        </div>
      </div>
    </div>
    <div
      className="card__accordion--up"
    >
      <div
        className="card__featureslistdropdown"
      >
        <h2>
          Features
        </h2>
        <div
          className="card__dropdown-divider"
        />
        <ul
          className="card__featureslistdropdown-unorderedlist"
        >
          <li>
            Local Agents
          </li>
          <li>
            Mobile Claim Reporting
          </li>
        </ul>
      </div>
      <div
        className="card__detailsdropdown"
      >
        <h2>
          Why 
          High Card
          ?
        </h2>
        <div
          className="card__dropdown-divider"
        />
        <p>
          Texas Farm Bureau offers auto insurance that best suits their customers' needs, with competitive rates and the right coverage for how and where you drive.
        </p>
      </div>
    </div>
    <button
      className="card__chevron-button"
      onClick={[Function]}
    >
      <svg
        className="card__chevron--down"
      >
        IconChevronDown.svg
      </svg>
    </button>
  </div>
  <div
    className="card"
    data-testid="card-test-id"
  >
    <div
      className="card__verified-mark"
    >
      <svg>
        IconStarWhite.svg
      </svg>
    </div>
    <div
      className="card__topfold-container"
    >
      <div
        className="card__topfold-subcontainer--1"
      >
        <div
          className="card__logo-container"
        >
          <img
            alt=""
            className="card__logo"
            src="logo-carrier-sm-default.png"
          />
        </div>
      </div>
      <div
        className="card__topfold-subcontainer--2"
      >
        <h1
          className="card__carrier-name"
        >
          Low Card
        </h1>
        <div
          className="card__star-icons"
        >
          <svg>
            IconStarGrey.svg
          </svg>
          <svg>
            IconStarGrey.svg
          </svg>
          <svg>
            IconStarGrey.svg
          </svg>
          <svg>
            IconStarGrey.svg
          </svg>
          <svg>
            IconStarGrey.svg
          </svg>
        </div>
        <div
          className="card__feature-icons"
        >
          <svg>
            IconFeaturesLocalAgent.svg
          </svg>
          <svg>
            IconFeaturesMobile.svg
          </svg>
        </div>
        <div
          className="card__carrier-tagline"
        >
          Austin, TX drivers could save up to 25% with Elephant®
        </div>
      </div>
      <div
        className="card__topfold-subcontainer--3"
      >
        <div
          className="card__verified-tag"
        >
          Verified Price
        </div>
        <div
          className="card__price"
        >
          <span
            className="card__price-dollar-sign"
          >
            $
          </span>
          NaN
           
          <span
            className="card__price-month"
          >
            / mo
          </span>
        </div>
      </div>
    </div>
    <div
      className="card__accordion--up"
    >
      <div
        className="card__featureslistdropdown"
      >
        <h2>
          Features
        </h2>
        <div
          className="card__dropdown-divider"
        />
        <ul
          className="card__featureslistdropdown-unorderedlist"
        >
          <li>
            Local Agents
          </li>
          <li>
            Mobile Claim Reporting
          </li>
        </ul>
      </div>
      <div
        className="card__detailsdropdown"
      >
        <h2>
          Why 
          Low Card
          ?
        </h2>
        <div
          className="card__dropdown-divider"
        />
        <p>
          Texas Farm Bureau offers auto insurance that best suits their customers' needs, with competitive rates and the right coverage for how and where you drive.
        </p>
      </div>
    </div>
    <button
      className="card__chevron-button"
      onClick={[Function]}
    >
      <svg
        className="card__chevron--down"
      >
        IconChevronDown.svg
      </svg>
    </button>
  </div>
</div>
`;
